### DISCUSSION TITLE
Layout has an invalid "generateMetadata" export:   Type "Props" is not valid.

### DISCUSSION THREAD

**Discussion by alamenai** [vercel/next.js#57767] [Help] [answered]
### Summary

**Good evening guys,**

I have an issue that I really could not figure out the reason behind, I want to create a dynamic title based on the route search parameters, so I used the `generateMetadata` function but I got this error each time  I run `npm run build`:

``` 
layout.tsx has an invalid "generateMetadata" export:
Type "Props" is not valid.

```

**This is my code:**

```tsx
import { AppConfig } from "@/config/app"
import { Metadata } from "next"
import React from "react"

const { app } = AppConfig

type Props = {
  searchParams: { [key: string]: string | string[] | undefined }
}

export async function generateMetadata({ searchParams }: Props): Promise<Metadata> {
  const title =
    searchParams?.modernization === "current"
      ? "Zusammenfassung des aktuellen energetischen Zustands"
      : "Zusammenfassung des energitschen Zustands nach der Modernisierung"

  return {
    title: `${app.energyCheck} | ${title}`,
  }
}

export default function Layout({ children }: { children: React.ReactNode }) {
  return <main>{children}</main>
}


```

**Comment by icyJoseph** [comment, marked as answer]
Pretty sure the issue is https://nextjs.org/docs/app/api-reference/functions/generate-metadata#returns:

- searchParams are only available in page.js segments.

If you remove it from the `type Props` alias, then it'll work, which confirms the above.

<img width="830" alt="Screenshot 2023-10-31 at 09 19 00" src="https://github.com/vercel/next.js/assets/21013447/3cb7a780-7314-48f8-baaa-fa9e34996163">

**Reply by alamenai** [reply]
Thank you so much @icyJoseph , yes I switched from layout.tsx to page.tsx and it works.
